dev/tools/gen_defaults/lib/list_tile_template.dart

```diff
@@ -0,0 +1,43 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'template.dart';
+
+class ListTileTemplate extends TokenTemplate {
+  const ListTileTemplate(super.blockName, super.fileName, super.tokens);
+
+  @override
+  String generate() => '''
+class _${blockName}DefaultsM3 extends ListTileThemeData {
+  const _${blockName}DefaultsM3(this.context)
+    : super(shape: ${shape("md.comp.list.list-item.container")});
+
+  final BuildContext context;
+
+  @override
+  Color? get tileColor =>  ${componentColor("md.comp.list.list-item.container")};
+
+  @override
+  TextStyle? get titleTextStyle => ${textStyle("md.comp.list.list-item.label-text")};
+
+  @override
+  TextStyle? get subtitleTextStyle => ${textStyle("md.comp.list.list-item.supporting-text")};
+
+  @override
+  TextStyle? get trailingAndLeadingTextStyle => ${textStyle("md.comp.list.list-item.trailing-supporting-text")};
+
+  @override
+  Color? get iconColor => MaterialStateColor.resolveWith((Set<MaterialState> states) {
```

Comment:
This is good, in the sense that it shows the way forward. Eventually it would be preferable if the ListTile and ListTileTheme color, iconColor, tileColor, textColor properties could all be MaterialStateColors. I don't know if we'll ever be able to phase out focusColor, hoverColor, splashColor, selectedTileColor ... but it would be nice to not have to use them.